src/compiler/crystal/tools/formatter.cr

```diff
@@ -1692,11 +1694,78 @@ module Crystal
 
       format_def_args node
 
-      format_macro_body node
+      if all_macro_literals?(node.body)
+        format_all_macro_literals(node, macro_node_line)
+      else
+        format_macro_body node
+      end
 
       false
     end
 
+    private def all_macro_literals?(node)
+      case node
+      when MacroLiteral
+        true
+      when Expressions
+        node.expressions.all? { |subnode| all_macro_literals?(subnode) }
+      else
+        false
+      end
+    end
```

Comment:
This seems to be defined below as `#only_macro_literal?(node)` already.

Reply:
Done, thanks!